pkg/reward/reward.go

```diff
@@ -70,12 +92,84 @@ func (self *Reward) Current(slot *big.Int, gasPriceEth1e18 *big.Int) (int64, err
 	return profitPercent, nil
 }
 
-func (self *Reward) GasUsed(slot *big.Int) (*big.Int, error) {
-	if gas, ok := self.gasUsed[slot.Int64()]; ok {
-		return gas, nil
+var tellorAddress = common.HexToAddress(contracts.TellorAddress)
+
+func (self *Reward) GasUsed(ctx context.Context, slot *big.Int) (*big.Int, error) {
+	ticker := time.NewTicker(1 * time.Second)
+
+	var (
+		vars struct {
+			Challenge  [32]byte
+			RequestIds [5]*big.Int
+			Difficutly *big.Int
+			Tip        *big.Int
+		}
+		err error
+	)
+	// Getting current challenge.
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, errors.New("context cancelled")
+		default:
+		}
+		vars, err = self.contractCaller.GetNewCurrentVariables(nil)
+		if err != nil {
+			level.Debug(self.logger).Log("msg", "calling GetNewCurrentVariables", "err", err)
+			<-ticker.C
+			continue
+		}
+		break
 	}
 
-	return nil, ErrNoDataForSlot{slot: slot.String()}
+	i := 0
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, errors.New("context cancelled")
+		default:
+		}
+		gasPrice, err := self.client.SuggestGasPrice(ctx)
+		if err != nil {
+			level.Debug(self.logger).Log("msg", "calling SuggestGasPrice", "err", err)
+			<-ticker.C
+			continue
+		}
+		reqVals := [5]*big.Int{
+			big.NewInt(math.MaxInt64),
+			big.NewInt(math.MaxInt64),
+			big.NewInt(math.MaxInt64),
+			big.NewInt(math.MaxInt64),
+			big.NewInt(math.MaxInt64),
+		}
+
+		packed, err := self.abi.Pack("submitMiningSolution", "", vars.RequestIds, reqVals)
+		if err != nil {
+			level.Debug(self.logger).Log("msg", "packing submitMiningSolution args", "err", err)
+			return nil, errors.Wrap(err, "packing submitMiningSolution args")
+		}
+		data := ethereum.CallMsg{
+			From:     self.account.Address,
+			To:       &tellorAddress,
+			GasPrice: gasPrice,
+			Data:     packed,
+		}
+		gasUsed, err := self.client.EstimateGas(ctx, data)
+		if err != nil {
+			level.Debug(self.logger).Log("msg", "calling EstimateGas", "err", err)
+			continue
+		}
+		if i%2 == 0 {
+			fmt.Println("gasLimit(random)", gasUsed)
+		} else {
+			fmt.Println("gasLimit(maxInt)", gasUsed)
+		}
+		i++
+		if i == 5 {
+			return nil, ErrNoDataForSlot{slot: slot.String()}
+		}
```

Comment:
this should never happen right? so we don't need it.

Reply:
Yeah, right!